tasks/main.yml

```
@@ -1,2 +1,44 @@
 ---
-# tasks file for skeleton
+- name: Load var file with package names based on the OS type
+  ansible.builtin.include_vars: "{{ lookup('first_found', params) }}"
+  vars:
+    params:
+      files:
+        - "{{ ansible_distribution }}_{{ ansible_distribution_release }}.yml"
+        - "{{ ansible_distribution }}.yml"
+        - "{{ ansible_os_family }}.yml"
+      paths:
+        - "{{ role_path }}/vars"
+
+- name: Install Samba client
+  ansible.builtin.package:
+    name: "{{ client_packages }}"
+
+- name: Install and enable Samba server
+  block:
+    - name: Install Samba server packages
+      ansible.builtin.package:
+        name: "{{ server_packages }}"
+    - name: Enable Samba service
+      ansible.builtin.service:
+        enabled: yes
+        name: "{{ service_name }}"
+    - name: Create Samba guest user
+      block:
+        - name: Create Samba guest user
+          ansible.builtin.user:
+            comment: Samba guest account
+            create_home: no
+            home: /nonexistent
+            name: "{{ guest_user }}"
+            shell: /sbin/nologin
+            system: yes
+        - name: Configure samba to use the guest account
```

Consistent capitalization:
```suggestion
        - name: Configure Samba to use the guest account
```